applications/GeoMechanicsApplication/tests/cpp_tests/custom_elements/test_line_interface_element.cpp

```diff
@@ -0,0 +1,540 @@
+    expected_left_hand_side(2, 6) = -ShearStiffness * 0.5;
+    expected_left_hand_side(3, 7) = -NormalStiffness * 0.5;
+    expected_left_hand_side(4, 0) = -ShearStiffness * 0.5;
+    expected_left_hand_side(5, 1) = -NormalStiffness * 0.5;
+    expected_left_hand_side(6, 2) = -ShearStiffness * 0.5;
+    expected_left_hand_side(7, 3) = -NormalStiffness * 0.5;
+
+    return expected_left_hand_side;
+}
+
+} // namespace
+
+namespace Kratos::Testing
+{
+
+using namespace Kratos;
+
+KRATOS_TEST_CASE_IN_SUITE(LineInterfaceElementIsAnElement, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    const LineInterfaceElement element;
+    auto                       p_casted_element = dynamic_cast<const Element*>(&element);
+    KRATOS_CHECK_NOT_EQUAL(p_casted_element, nullptr);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(LineInterfaceElementCanCreateInstanceWithGeometryInput, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const LineInterfaceElement element;
+    const auto p_geometry   = std::make_shared<LineInterfaceGeometry2D2Plus2Noded>(CreateNodes());
+    const auto p_properties = std::make_shared<Properties>();
+
+    // Act
+    const auto p_created_element = element.Create(1, p_geometry, p_properties);
+
+    // Assert
+    EXPECT_NE(p_created_element, nullptr);
+    EXPECT_EQ(p_created_element->Id(), 1);
+    EXPECT_NE(p_created_element->pGetGeometry(), nullptr);
+    EXPECT_NE(p_created_element->pGetProperties(), nullptr);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(LineInterfaceElementCanCreateInstanceWithNodeInput, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const auto nodes        = CreateNodes();
+    const auto p_properties = std::make_shared<Properties>();
+
+    // The source element needs to have a geometry, otherwise the version of the
+    // Create method with a node input will fail.
+    const auto p_geometry = std::make_shared<LineInterfaceGeometry2D2Plus2Noded>(nodes);
+    const LineInterfaceElement element(0, p_geometry, p_properties);
+
+    // Act
+    const auto p_created_element = element.Create(1, nodes, p_properties);
+
+    // Assert
+    EXPECT_NE(p_created_element, nullptr);
+    EXPECT_EQ(p_created_element->Id(), 1);
+    EXPECT_NE(p_created_element->pGetGeometry(), nullptr);
+    EXPECT_NE(p_created_element->pGetProperties(), nullptr);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(LineInterfaceElement_ReturnsTheExpectedDoFList, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    // Arrange
+    const auto p_properties = std::make_shared<Properties>();
+
+    Model model;
+    const auto element = CreateHorizontalUnitLength2Plus2NodedLineInterfaceElementWithUDofs(model, p_properties);
+
+    // Act
+    const auto              dummy_process_info = ProcessInfo{};
+    Element::DofsVectorType degrees_of_freedom;
+    element.GetDofList(degrees_of_freedom, dummy_process_info);
+
+    // Assert
+    KRATOS_EXPECT_EQ(degrees_of_freedom.size(), 8);
+    for (std::size_t i = 0; i < degrees_of_freedom.size(); i += 2) {
```

Comment:
remaining raw loop?

Reply:
Indeed, we didn't see a more readable way or a standard algorithm to replace this unfortunately (due to the `i+=2`). We could create a `std::vector<Variable<double>> expected_degrees_of_freedom`, but then we still need the loop to compare the ith element:
```
for (std::size_t i = 0; i < degrees_of_freedom.size(); ++i) {
        KRATOS_EXPECT_EQ(degrees_of_freedom[i]->GetVariable(), expected_degrees_of_freedom[i]);
}
```